main.go

```diff
@@ -1,62 +1,284 @@
 package main
 
 import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
 	"fmt"
+	"io/ioutil"
 	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
 
+	"github.com/sensu-community/sensu-plugin-sdk/httpclient"
 	"github.com/sensu-community/sensu-plugin-sdk/sensu"
 	"github.com/sensu/sensu-go/types"
 )
 
-type HandlerConfig struct {
+// Handler represents the sensu-puppet-handler plugin
+type Handler struct {
 	sensu.PluginConfig
-	Example string
+	endpoint                 string
+	puppetCert               string
+	puppetKey                string
+	puppetCACert             string
+	puppetInsecureSkipVerify bool
+	sensuAPIURL              string
+	sensuAPIKey              string
+	sensuCACert              string
 }
 
-type ConfigOptions struct {
-	Example sensu.PluginConfigOption
-}
+const (
+	defaultAPIPath      = "pdb/query/v4/nodes"
+	labelPuppetNodeName = "puppet_node_name"
+)
 
 var (
-	handlerConfig = HandlerConfig{
+	handler = Handler{
 		PluginConfig: sensu.PluginConfig{
 			Name:     "sensu-puppet-handler",
 			Short:    "Deregister Sensu entities without an associated Puppet node",
-			Timeout:  10,
 			Keyspace: "sensu.io/plugins/sensu-puppet-handler/config",
 		},
 	}
 
-	handlerConfigOptions = ConfigOptions{
-		Example: sensu.PluginConfigOption{
-			Path:      "example",
-			Env:       "HANDLER_EXAMPLE",
-			Argument:  "example",
+	options = []*sensu.PluginConfigOption{
+		&sensu.PluginConfigOption{
+			Path:      "endpoint",
+			Env:       "PUPPET_ENDPOINT",
+			Argument:  "endpoint",
 			Shorthand: "e",
-			Default:   "",
-			Usage:     "An example configuration option",
-			Value:     &handlerConfig.Example,
+			Usage:     "the PuppetDB API endpoint (URL). If an API path is not specified, /pdb/query/v4/nodes/ will be used",
+			Value:     &handler.endpoint,
+		},
+		&sensu.PluginConfigOption{
+			Path:     "cert",
+			Env:      "PUPPET_CERT",
+			Argument: "cert",
+			Usage:    "path to the SSL certificate PEM file signed by your site's Puppet CA",
+			Value:    &handler.puppetCert,
+		},
+		&sensu.PluginConfigOption{
+			Path:     "key",
+			Env:      "PUPPET_KEY",
+			Argument: "key",
+			Usage:    "path to the private key PEM file for that certificate",
+			Value:    &handler.puppetKey,
+		},
+		&sensu.PluginConfigOption{
+			Path:     "cacert",
+			Env:      "PUPPET_CACERT",
```

Comment:
Similar to the suggestion above
```suggestion
			Env:      "PUPPET_CA_CERT",
```